src/store/plugins/load-geojson-style-and-data.plugin.js

```diff
@@ -36,13 +68,19 @@ async function loadDataAndStyle(geoJsonLayer) {
         clone.geoJsonData = data
         clone.geoJsonStyle = style
         clone.isLoading = false
-        return clone
+        await store.dispatch('updateLayer', {
+            layer: clone,
+            ...dispatcher,
+        })
     } catch (error) {
         log.error(`Error while fetching GeoJSON data/style for layer ${geoJsonLayer?.id}`, error)
         clone.isLoading = false
         clone.errorKey = 'loading_error_network_failure'
         clone.hasError = true
-        return clone
+        await store.dispatch('updateLayer', {
+            layer: clone,
+            ...dispatcher,
+        })
```

Comment:
Here I just modified the code to avoid multiple dispatch when we have more than one geojson layer, would be good to keep it for startup performance.